nixos/modules/services/misc/gitea.nix

```diff
@@ -478,6 +505,47 @@ in
       ];
     };
 
+    services.nginx = lib.mkIf cfg.configureNginx {
+      enable = true;
+      virtualHosts."${cfg.settings.server.DOMAIN}" = {
+        root = "/var/empty";
+
+        locations."/" = {
+          proxyPass = if (cfg.settings.server.PROTOCOL == "http+unix") then "http://unix:${cfg.settings.server.HTTP_ADDR}" else "http://127.0.0.1:${toString cfg.httpPort}";
```

Comment:
When deploying this, I encountered this error:

```
       error: The option `services.nginx.virtualHosts."git.supersandro.de".locations."/".proxyPass' has conflicting definition values:
       - In `/nix/store/7z411i6gb3szhc9hxbiilr6plmxsml9s-source/machines/hydrogen/configuration.nix': "http://gitea"
       - In `/nix/store/shb47577aqvq18559ic15g83379fajy6-source/nixos/modules/services/misc/gitea.nix': "http://unix:/run/gitea/gitea.sock"
```

which I think is acceptable after activating the option but I think we could switch to upstreams but I am totally fine with moving this into another PR.

Reply:
Replaced to use nginx upstreams option.